Makefile

```diff
@@ -37,7 +37,9 @@ bindata: ## Update project bindata
 .PHONY: fix
 fix: ## Fixup files in the repo.
 	go mod tidy
-	go fmt ./...
+	# the golangci-lint run --fix automatically fix issues when the linter/check supports auto-fix
+	# Note: only gofmt, gosimple and goimport checks used supports auto-fix
+	./tools/scripts/fetch golangci-lint 1.31.0 && ./tools/bin/golangci-lint run --fix
```

Comment:
Including this in the `fix` target means that it will be run as part of our sanity test and any changes will be rejected if there's a diff.

Are we absolutely sure we want to put this in our critical path for merging PRs? I would vote against this change.

Reply:
Are we calling the make fix in the test-sanity? if yes, I do not think that we should do that.

Reply:
if has a lint issue the test-sanity will fail anyway. 
The  `run --fix` only fix issues whcih would be raised and make the lint fails such as when it is missing space at the end of the file.

Reply:
Was the original intent of `fix` to define some basic standards we want to see across the codebase? If so, then this should be part of sanity tests.

Reply:
The intent of `make fix` was to separate out the pieces of the sanity tests that apply expected changes that we want to enforce into a separate make target. For a very long time now, those fixes have been `go mod tidy` and `go fmt ./...`. That way a contributor could run `make fix`, review any changes, and commit them prior to submitting a PR.

With this change we would apply and enforce all automated fixes of all enabled linters. If we add a linter in the future that supports autofixes, that would become part of the sanity test.

If that's the explicit direction we want to go, that's fine. I just wanted to highlight this as a concern I have.